### DISCUSSION TITLE
[v3] Layout slots are not rendered.

### DISCUSSION THREAD

**Discussion by PhiloNL** [livewire/livewire#5846] [7. Bugs (archived)]
Moving issue over: https://github.com/livewire/next/issues/93
Layout slots don't seem to be rendered inside Livewire components. The string `Hello World!!` is never rendered.

**Component**
```php
<?php

namespace App\Livewire;

use Livewire\Component;

class Login extends Component
{
    public function render()
    {
        return view('livewire.login');
    }
}

```

**livewire.login.blade.php**
```blade
<x-slot name="header">
    Hello World!!
</x-slot>

<div>
    Foobar
</div>
```

**components/layouts/app.blade.php**
```Blade
<html>
<head>
    <title>Livewire</title>

    @livewireStyles
</head>

<body>

    {{ $header ?? 'no header' }}

    {{ $slot }}

    @livewireScripts
</body>
</html>
```

While a regular blade file does work, for example:

**route**
```php
Route::view('login', 'livewire.login');
```

**login.blade.php**
```blade
<x-layouts.app>
    <x-slot name="header">
        Hello World!!
    </x-slot>

    <div>
        Foobar
    </div>

</x-layouts.app>
```

**Comment by PerryvanderMeer** [comment]
This might be related to #5818?

**Comment by PhiloNL** [comment]
Yes, in v3 this doesn't work yet and I think Caleb is looking into it.

**Comment by joshhanley** [comment]
@PhiloNL any chance you can submit a failing test for this? It worked in V2 so we should be able to get it working again.

**Reply by PhiloNL** [reply]
Added :) https://github.com/livewire/livewire/pull/5962

**Comment by schonhoff** [comment]
It is not only slots. @push directive or @sections directive doesn't work as well. (See https://github.com/livewire/livewire/discussions/5818 )

Edit:

Hello,

after some testing on V3 I noticed, that my section/yield/push blade directives and additional slots are not working anymore. I don't want to duplicate my layout and change it for the Livewire routes, because I have some routes without Livewire and some with Livewire. If I need to do it, it would add extra maintenance. It worked in V2 with the following component/layout:

V2 Component:
```blade
<div>
  component stuff
</div>
@section('title', 'My new Title')
@push('js-files')
  <script src="https://www.google.com/recaptcha/api.js" async defer></script>
@endpush
<x-slot:inlineScripts>
  <script type="text/javascript">
    {{-- needs to be var --}}
    var onRecaptchaCallback = function () {
      $set('recaptcha', grecaptcha.getResponse());
    };
  </script>
</x-slot:inlineScripts>
```

V2 layout.app.blade.php
```blade
<!DOCTYPE html>
<html lang="{{ app()->getLocale() }}" class="scroll-smooth">
<head>
  <title>@yield('title', config('app.name', 'T-Kfz'))</title>
  @stack('js-files')
  [...]
  @livewireStyles
</head>
<body>
  {{ $slot }}
  @livewireScript
  @isset($inlineScripts)
    {{ $inlineScripts }}
  @endisset
  [...]
</body>
```

After migrating to V3 I changed nothing to this component but the title, the js-file and the inline script isn't set anymore. Isn't it possible for Livewire to use this blade directives and slots anymore? The js-file is very important because of external js-libraries used in some Full-Page Livewire components and does need to set the js file of this in the header (for example google recaptcha).

Thanks for all and have a nice day :-)

**Reply by joshhanley** [reply]
@schonhoff can you copy your comment from over there to here, so it's all contained in one place? Thanks!

**Reply by schonhoff** [reply]
Done :-) I updated the first comment.

**Reply by joshhanley** [reply]
@schonhoff thanks! Much appreciated 🙂

**Comment by nam-co** [comment]
This seems like an important part of beta.4, just in case this is also not working but worked on v2
```
return view('livewire.somepage-wire')
        ->extends('layouts.backend')
	->layoutData(['user'=>$this->user'])
         ->section('content');
```

**Reply by joshhanley** [reply]
@nam-co not sure what you mean by "an important part of beta.4"? This is a bug that hasn't been fixed yet, so it's not in beta.4. There is now a failing test PR #5962 which will help with the investigation 🙂

**Comment by samharvey44** [comment]
Thought I was going crazy and doing something wrong! Glad it's not just me.

**Comment by danielarcher** [comment]
So this is still a bug on main branch, correct? I see that you guys merged the test cases at https://github.com/livewire/livewire/pull/5962

**Reply by samharvey44** [reply]
It seems to be half fixed to me. If you explicitly tell your livewire component which layout to use with `->layout()` in your component's render method, it works. However, if you don't do this, and you let Livewire determine the layout automatically, it doesn't work.

I made a discussion for it here - https://github.com/livewire/livewire/discussions/6132

**Comment by utqinadhif** [comment]
When the version 3 can be stable master version? Can i use version 3 for production?

**Comment by Crinchy** [comment]
I still having issues in v3.0.1 with the layouts, after logging into my web-application its not changing the app layout correctly. It still uses my guest layout.

**Comment by xyNNN** [comment]
Whats the current state of this topic?

**Reply by DanielUgalde95** [reply]
It works perfectly if you remove lazy loading...

Is there a solution to add slots without removing lazy load from the component?
